ql/src/java/org/apache/hadoop/hive/ql/optimizer/ppr/PartitionPruner.java

```diff
@@ -139,6 +140,20 @@ public static boolean onlyContainsPartnCols(Table tab, ExprNodeDesc expr) {
     return true;
   }
 
+  public static boolean isPartitioned(Table table){
+    if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
+      return table.getStorageHandler().isPartitioned(table);
+    } else {
+      return table.isPartitioned();
+    }
+  }
+  
+  public static boolean isPartitionKey(Table table, String columnName){
+    List<String> partitionKeyNames = table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned() ?
+        table.getStorageHandler().getAllPartitionKeyNames(table) : table.getPartCols().stream().map(FieldSchema::getName).collect(
```

Comment:
table. has a method `getPartColNames`

Reply:
It gives back an empty list for iceberg tables.